sql/src/main/java/io/crate/action/sql/SessionContext.java

```diff
@@ -58,14 +60,21 @@ public String defaultSchema() {
     }
 
     @Nullable
-    public String userName() {
-        return userName;
+    public User user() {
+        return user;
     }
 
     public void setDefaultSchema(@Nullable String schema) {
         defaultSchema = MoreObjects.firstNonNull(schema, Schemas.DEFAULT_SCHEMA_NAME);
     }
 
+    /**
+     * set an authenticated user
+     */
+    public void setUser(User user) {
```

Comment:
I liked having the User in the constructor better, why did you change  this to a setter? Afaik it doesn't make sense to change the user on a session?